frontends/web/src/routes/bitsurance/widget.tsx

```diff
@@ -93,24 +93,42 @@ export const BitsuranceWidget = ({ code }: TProps) => {
       version: MessageVersion.V0,
       type: V0MessageType.Address,
       bitcoinAddress: address,
+      extendedPublicKey: xpub,
       signature: sig,
     });
 
     current.contentWindow?.postMessage(message, '*');
   };
 
+  const getWPKHxPub = () => {
+    let wpkhConfig = accountInfo?.signingConfigurations.find(config =>
+      config.bitcoinSimple?.scriptType === 'p2wpkh'
+    );
+    return wpkhConfig?.bitcoinSimple?.keyInfo.xpub;
+  };
+
   const handleRequestAddress = (message: RequestAddressV0Message) => {
     signing = true;
     const addressType = message.withScriptType ? String(message.withScriptType) : '';
     const withMessageSignature = message.withMessageSignature ? message.withMessageSignature : '';
+    const withExtendedPublicKey = !!message.withExtendedPublicKey;
     signAddress(
       addressType,
       withMessageSignature,
       code)
       .then(response => {
         signing = false;
         if (response.success) {
-          sendAddress(response.address, response.signature);
+          if (withExtendedPublicKey) {
+            const xpub = getWPKHxPub();
```

Comment:
Could you make the function more generic using the script type as a parameter? So that we don't need updates in the code to support other script types in the future.

Reply:
OK.